src/libs/migrations/NVPMigration.ts

```diff
@@ -0,0 +1,84 @@
+import after from 'lodash/after';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+const migrations = {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    nvp_lastPaymentMethod: ONYXKEYS.NVP_LAST_PAYMENT_METHOD,
+    isFirstTimeNewExpensifyUser: ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER,
+    preferredLocale: ONYXKEYS.NVP_PREFERRED_LOCALE,
+    preferredEmojiSkinTone: ONYXKEYS.PREFERRED_EMOJI_SKIN_TONE,
+    frequentlyUsedEmojis: ONYXKEYS.FREQUENTLY_USED_EMOJIS,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    private_blockedFromConcierge: ONYXKEYS.NVP_BLOCKED_FROM_CONCIERGE,
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    private_pushNotificationID: ONYXKEYS.NVP_PRIVATE_PUSH_NOTIFICATION_ID,
+    tryFocusMode: ONYXKEYS.NVP_TRY_FOCUS_MODE,
+    introSelected: ONYXKEYS.NVP_INTRO_SELECTED,
+    hasDismissedIdlePanel: ONYXKEYS.NVP_HAS_DISMISSED_IDLE_PANEL,
+};
+
+// This migration changes the keys of all the NVP related keys so that they are standardized
+export default function () {
+    return new Promise<void>((resolve) => {
+        // We add the number of manual connections we add below
+        const resolveWhenDone = after(Object.entries(migrations).length + 2, () => resolve());
+
+        for (const [oldKey, newKey] of Object.entries(migrations)) {
+            const connectionID = Onyx.connect({
+                // @ts-expect-error oldKey is a variable
+                key: oldKey,
+                callback: (value) => {
+                    Onyx.disconnect(connectionID);
+                    if (value !== null) {
```

Comment:
NAB: I'm kind of surprised that there aren't lint warnings to switch these to early returns. I guess it's because there is no `return` statement, but I think it would be nice to follow the style of early returns so the code style is consistent.

Reply:
I like this approach more because if I do early return I will need to call `resolveWhenDone` twice

Reply:
Disregard, changing this given I need to pass resolveWhenDone to the promise too